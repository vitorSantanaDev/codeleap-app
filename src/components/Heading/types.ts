import { ReactNode } from 'react'

export type HeadingProps = {
  children: ReactNode
  color?: 'white' | 'black' | 'gray'
  size?: 'medium' | 'xlarge'
}
