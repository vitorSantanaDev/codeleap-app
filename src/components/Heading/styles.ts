import styled, { css } from 'styled-components'
import { HeadingProps } from './types'

type WrapperProps = Pick<HeadingProps, 'color' | 'size'>

export const Wrapper = styled.h2<WrapperProps>`
  ${({ theme, color, size }) => css`
    color: ${theme.colors[color!]};
    font-weight: ${theme.font.bold};
    font-size: calc(${theme.font.sizes[size!]} + 0.2rem);
  `}
`
