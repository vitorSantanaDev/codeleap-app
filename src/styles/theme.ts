export default {
  grid: {
    container: '130rem'
  },
  border: {
    radius: '0.8rem'
  },
  font: {
    family:
      "Poppins, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif",
    normal: 400,
    bold: 700,
    sizes: {
      xsmall: '1.2rem',
      small: '1.4rem',
      medium: '1.6rem',
      large: '1.8rem',
      xlarge: '2.0rem',
      xxlarge: '2.8rem',
      huge: '5.2rem'
    }
  },
  colors: {
    primary: '#7695EC',
    white: '#FFFFFF',
    black: '#000000',
    gray: '#999999',
    dark_gray: '#777777',
    light_gray: '#CCCCCC',
    red: '#FF5151',
    green: '#47B960',
    overlay_color: 'rgba(119, 119, 119, 0.8)'
  },
  spacings: {
    xxsmall: '0.8rem',
    xsmall: '1.6rem',
    small: '2.4rem',
    medium: '3.2rem',
    large: '4.0rem',
    xlarge: '4.8rem',
    xxlarge: '5.6rem'
  },
  layers: {
    base: 10,
    menu: 20,
    overlay: 30,
    modal: 40,
    alwaysOnTop: 50
  },
  transition: {
    fast: '0.1s ease-in-out',
    default: '0.3s ease-in-out'
  }
} as const
