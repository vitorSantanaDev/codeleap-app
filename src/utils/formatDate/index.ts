import {
  format,
  parseISO,
  minutesToHours,
  secondsToMinutes,
  differenceInSeconds
} from 'date-fns'

export const formattingStringWithTheTimePassedBetweenTwoDate = (
  hours: number,
  minutes: number,
  seconds: number
): string => {
  if (hours < 1 && minutes < 1) {
    return `${seconds} seconds ago`
  }

  let formattedTime = ''

  if (hours > 0) {
    formattedTime = `${hours} ${hours > 1 ? 'hours' : 'hour'} ago`
  } else if (hours < 1) {
    formattedTime = `${minutes} minutes ago`
  }

  return formattedTime
}

export const rangeOfTimePassedBetweenTwoDates = (
  start: string,
  end: string
): string => {
  const startDate = parseISO(start)
  const endDate = parseISO(end)

  const allSeconds = differenceInSeconds(endDate, startDate)
  const allMinutes = secondsToMinutes(allSeconds)

  const minutesInSeconds = allMinutes * 60
  const seconds = allSeconds - minutesInSeconds

  const hours = minutesToHours(allMinutes)

  const hoursInMinutes = hours * 60

  const minutes = allMinutes - hoursInMinutes

  return hours > 24
    ? format(new Date(startDate), 'dd/MM/yyyy')
    : formattingStringWithTheTimePassedBetweenTwoDate(hours, minutes, seconds)
}
