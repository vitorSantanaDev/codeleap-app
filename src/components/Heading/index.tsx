import React from 'react'

import { HeadingProps } from './types'

import * as S from './styles'

const Heading: React.FC<HeadingProps> = ({
  children,
  color = 'black',
  size = 'medium'
}) => {
  return (
    <S.Wrapper color={color} size={size}>
      {children}
    </S.Wrapper>
  )
}

export default Heading
