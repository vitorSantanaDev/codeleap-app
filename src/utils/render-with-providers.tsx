import { Provider } from 'react-redux'
import React, { PropsWithChildren } from 'react'

import theme from 'styles/theme'
import { ThemeProvider } from 'styled-components'
import { RenderOptions, render } from '@testing-library/react'

import { RootState } from 'redux/store'
import { userReducer } from 'redux/actions/user-slice'
import { PreloadedState, configureStore } from '@reduxjs/toolkit'
import { QueryClient, QueryClientProvider } from 'react-query'

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  initialState?: PreloadedState<RootState>
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  store?: any
}

export const renderWithProviders = (
  ui: React.ReactElement,
  {
    initialState = {},
    store = configureStore({
      reducer: { user: userReducer },
      preloadedState: initialState
    }),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) => {
  function Wrapper({ children }: PropsWithChildren<unknown>): JSX.Element {
    return (
      <QueryClientProvider client={new QueryClient()}>
        <Provider store={store}>
          <ThemeProvider theme={theme}>{children}</ThemeProvider>
        </Provider>
      </QueryClientProvider>
    )
  }

  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}
