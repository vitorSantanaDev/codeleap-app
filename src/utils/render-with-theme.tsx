import React, { ReactElement } from 'react'

import theme from 'styles/theme'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

export const renderWithTheme = (component: ReactElement) =>
  render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
