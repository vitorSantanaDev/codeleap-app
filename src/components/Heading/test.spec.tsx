import React from 'react'
import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/render-with-theme'

import theme from 'styles/theme'
import Heading from '.'

describe('<Heading />', () => {
  it('should render heading with default color and default size', () => {
    renderWithTheme(<Heading>Heading</Heading>)

    const heading = screen.getByRole('heading', { name: /heading/i })
    expect(heading).toBeInTheDocument()

    expect(heading).toHaveStyle({
      color: theme.colors.black,
      fontSize: `calc(${theme.font.sizes.medium} + 0.2rem)`
    })
  })

  it('should render heading with color white and large size', () => {
    renderWithTheme(
      <Heading color="white" size="xlarge">
        Heading
      </Heading>
    )

    const heading = screen.getByRole('heading', { name: /heading/i })

    expect(heading).toHaveStyle({
      color: theme.colors.white,
      fontSize: `calc(${theme.font.sizes.xlarge} + 0.2rem)`
    })
  })

  it('should render heading with color gray', () => {
    renderWithTheme(<Heading color="gray">Heading</Heading>)

    const heading = screen.getByRole('heading', { name: /heading/i })

    expect(heading).toHaveStyle({
      color: theme.colors.gray
    })
  })

  it('should match snapshot', () => {
    const { container } = renderWithTheme(<Heading>Heading</Heading>)
    expect(container.firstChild).toMatchSnapshot()
  })
})
