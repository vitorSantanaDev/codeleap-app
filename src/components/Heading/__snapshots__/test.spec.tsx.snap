// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should match snapshot 1`] = `
.c0 {
  color: #000000;
  font-weight: 700;
  font-size: calc(1.6rem + 0.2rem);
}

<h2
  class="c0"
  color="black"
>
  Heading
</h2>
`;
